File: web/src/model/weatherCode.ts
/**
 * WMO weather-code presentation.
 *
 * Ported 1:1 from `model/WeatherCode.kt`.
 */

export interface WeatherCodePresentation {
  readonly label: string;
  readonly shortLabel: string;
}

export function presentWeatherCode(code: number): WeatherCodePresentation {
  switch (code) {
    case 0:
      return { label: "Clear sky", shortLabel: "Clear" };
    case 1:
    case 2:
    case 3:
      return { label: "Partly cloudy", shortLabel: "Cloudy" };
    case 45:
    case 48:
      return { label: "Fog", shortLabel: "Fog" };
    case 51:
    case 53:
    case 55:
    case 56:
    case 57:
      return { label: "Drizzle", shortLabel: "Drizzle" };
    case 61:
    case 63:
    case 65:
    case 66:
    case 67:
      return { label: "Rain", shortLabel: "Rain" };
    case 71:
    case 73:
    case 75:
    case 77:
      return { label: "Snow", shortLabel: "Snow" };
    case 80:
    case 81:
    case 82:
      return { label: "Rain showers", shortLabel: "Showers" };
    case 85:
    case 86:
      return { label: "Snow showers", shortLabel: "Snow" };
    case 95:
    case 96:
    case 99:
      return { label: "Thunderstorm", shortLabel: "Storm" };
    default:
      return { label: "Unknown weather", shortLabel: "Unknown" };
  }
}

File: shared/src/commonMain/kotlin/com/cloudbasepredictor/model/WeatherCode.kt
package com.cloudbasepredictor.model

data class WeatherCodePresentation(
    val label: String,
    val shortLabel: String,
)

enum class WeatherCondition {
    CLEAR_SKY,
    PARTLY_CLOUDY,
    FOG,
    DRIZZLE,
    RAIN,
    SNOW,
    RAIN_SHOWERS,
    SNOW_SHOWERS,
    THUNDERSTORM,
    UNKNOWN,
}

object WeatherCode {
    fun condition(code: Int): WeatherCondition = when (code) {
        0 -> WeatherCondition.CLEAR_SKY
        1, 2, 3 -> WeatherCondition.PARTLY_CLOUDY
        45, 48 -> WeatherCondition.FOG
        51, 53, 55, 56, 57 -> WeatherCondition.DRIZZLE
        61, 63, 65, 66, 67 -> WeatherCondition.RAIN
        71, 73, 75, 77 -> WeatherCondition.SNOW
        80, 81, 82 -> WeatherCondition.RAIN_SHOWERS
        85, 86 -> WeatherCondition.SNOW_SHOWERS
        95, 96, 99 -> WeatherCondition.THUNDERSTORM
        else -> WeatherCondition.UNKNOWN
    }

    /** Deterministic English fallback used by shared previews and non-localized callers. */
    fun present(code: Int): WeatherCodePresentation = when (condition(code)) {
        WeatherCondition.CLEAR_SKY -> WeatherCodePresentation(label = "Clear sky", shortLabel = "Clear")
        WeatherCondition.PARTLY_CLOUDY -> WeatherCodePresentation(label = "Partly cloudy", shortLabel = "Cloudy")
        WeatherCondition.FOG -> WeatherCodePresentation(label = "Fog", shortLabel = "Fog")
        WeatherCondition.DRIZZLE -> WeatherCodePresentation(label = "Drizzle", shortLabel = "Drizzle")
        WeatherCondition.RAIN -> WeatherCodePresentation(label = "Rain", shortLabel = "Rain")
        WeatherCondition.SNOW -> WeatherCodePresentation(label = "Snow", shortLabel = "Snow")
        WeatherCondition.RAIN_SHOWERS -> WeatherCodePresentation(label = "Rain showers", shortLabel = "Showers")
        WeatherCondition.SNOW_SHOWERS -> WeatherCodePresentation(label = "Snow showers", shortLabel = "Snow")
        WeatherCondition.THUNDERSTORM -> WeatherCodePresentation(label = "Thunderstorm", shortLabel = "Storm")
        WeatherCondition.UNKNOWN -> WeatherCodePresentation(label = "Unknown weather", shortLabel = "Unknown")
    }
}
